### ISSUE TITLE
v3: ReactQueryDevtools initialIsOpen={false} Ignored

### ISSUE THREAD

**Issue by DaleyKD** [tannerlinsley/react-query#1394]
## Describe the bug
When setting `initialIsOpen` to `false`, the `ReactQueryDevtools` stays open.

## Expected behavior
**I realize this is probably because of the local storage.**
We may disagree on this expected behavior, but I would kindly "argue" that if I set it to "false" in my code, then that trumps the local storage.

## Side note
The typings for `initialIsOpen` is `any` when it should be `boolean`.

**Comment by TkDodo** [comment]
It seems that:
- you set it to `initialIsOpen=false`.
- you open the devtools.
- you close the page and come back.

--> the devtools are now open.

is that the case?

if so, then I think this is indeed the desired behaviour, as the last selected user selection is respected. The name is `initialIsOpen`, which indicates an initial value if no user selection has been made imo.

**Comment by DaleyKD** [comment]
That's one use case and yes, you're correct.

The one that sparked my investigation was:
* I initially have `initialIsOpen=true`
* I go about my testing with the devtools open
* I realize, man, I don't want this thing open when I start this app up every time.  I'd rather it be closed by default, every time.  I'd rather the user have to open it if they want to use it.
* I set it to `initialIsOpen=false`
* Hmmm... it's still open.

Then I saw the local storage and realized that's the reason.  And I guess my interpretation of "initial" was a little different.  I can't argue enough, though. :)  I mean, it IS just devtools. :D
